app/components/ui/menu.tsx

```diff
@@ -0,0 +1,48 @@
+import { Menu, Transition } from '@headlessui/react';
+import { type ReactElement } from 'react';
+import { UserCircleIcon } from '~/components/ui/icons';
+import { Link } from '~/components/ui/links';
+import { LogoutButton } from '~/components/layout/top-nav';
+
+export function TopNavUserMenu() {
+  return (
+    <StyledMenu
+      button={
+        <div className="w-8">
+          <UserCircleIcon />
+        </div>
+      }
+    >
+      <>
+        <Menu.Item>
+          <Link to="/settings">Settings</Link>
+        </Menu.Item>
+        <Menu.Item>
+          <LogoutButton />
+        </Menu.Item>
+      </>
```

Comment:
Just curious, is this fragment required?

Reply:
yea, as the the children of StyledMenu are expecting a single element